### ISSUE TITLE
(occasional) core dump during `treeSeqOutput`

### ISSUE THREAD

**Issue by clwgg** [MesserLab/SLiM#272]
I am (occasionally) experiencing core dump failures in a simulation setup that uses tree sequence recording, and where the simulation is initialized from a tree sequence generated with msprime and pyslim.

The entire simulation setup is a little involved, but I've tried to narrow it down to a fairly minimal example that still experiences the failure by cutting out a bunch of features. The remaining features in this minimal example that may (or may not) be important are a nonWF setup with age structure, remembered individuals at several time points, as well as eventual extinction of the population:

``` R
initialize() {
  if (!exists("StartTree")) {
    defineConstant("StartTree", "STARTING_min.trees");
  }

  defineConstant("maxK", 300);
  defineConstant("L", 1e8);

  defineConstant("BaseFn", "TMP-initTS-MANUAL");
  defineConstant("outfn", BaseFn + ".raw.trees");

  defineConstant("SampleSize", 300);

  defineConstant("MaxAge", 10);
  defineConstant("FertAge", 3);
  defineConstant("Fertility", 0.5);

  defineConstant("gremember", asInteger(c(2000 + seq(1, 1000, 100))));

  initializeSLiMModelType("nonWF");
  initializeTreeSeq();
  initializeMutationType("m1", 0.5, "n", 0.0, 0.1);
  initializeGenomicElementType("g1", m1, 1.0);
  initializeGenomicElement(g1, 0, L-1);
  initializeRecombinationRate(1e-8);
  initializeMutationRate(1e-9);

}

reproduction() {
  if (sim.generation > 2500) {
    return;
  }

  litterSize = rpois(1, Fertility);
  if (litterSize & (individual.age >= FertAge)) {
    mate = subpop.sampleIndividuals(1, minAge=FertAge);

    for (i in 1:litterSize)
      subpop.addCrossed(individual, mate);
  }
}

1 early() {
  sim.rescheduleScriptBlock(s1, generations=gremember);
  sim.readFromPopulationFile(StartTree);
}

early() {
  inds = sim.subpopulations.individuals;
  inds.fitnessScaling = maxK / length(inds);
  age_mortality = sqrt((MaxAge - inds.age) / MaxAge);
  inds.fitnessScaling = inds.fitnessScaling * age_mortality;
}

fitness(m1) {
  return 1.0;
}

late() {
  if (p0.individualCount == 0) {
    catn("Extinction in generation " + sim.generation + ".");
    sim.treeSeqOutput(outfn);
    sim.simulationFinished();
  }
}

s1 late() {
  catn("Saving generation: " + sim.generation);
  sim.treeSeqRememberIndividuals(sample(p0.individuals, pmin(SampleSize, p0.individualCount)));
}
```

A starting tree can be generated with:
``` python
import msprime
import pyslim

N = 300
Ne = N
L = 1e8
RecombRate = 1e-8
MutRate = 1e-9

demog_model = msprime.Demography()
demog_model.add_population(initial_size=Ne)
ts = msprime.sim_ancestry(samples=N, demography=demog_model,
                          sequence_length=L, recombination_rate=RecombRate)

ts = pyslim.annotate_defaults(ts, model_type="nonWF", slim_generation=1)

ts = msprime.sim_mutations(ts, rate=(MutRate), keep=True,
                           model=msprime.SLiMMutationModel(type=1))

print(ts)
ts.dump("STARTING_min.trees")
```
but I also attach one here: [STARTING_min.trees.zip](https://github.com/MesserLab/SLiM/files/7893604/STARTING_min.trees.zip)
(github wouldn't let me upload a `.trees` file, so I added a .zip ending without actually zipping it... `mv STARTING_min.trees.zip STARTING_min.trees`).

As far as I can tell, the core dump occurs in `SLiMSim::WriteTreeSequence` during tree sequence output, right after the call to `AddParentsColumnForOutput` [here](https://github.com/MesserLab/SLiM/blob/master/core/slim_sim.cpp#L7357).

Calling `slim -s 22001244111718 minimal_error.slim` (with `STARTING_min.trees` being generated and the above script saved as `minimal_error.slim`) reproduced the error for me on two different machines, with both the SliM 3.7 release as well as github development head version. For the python part, I am on:
```
msprime                 1.1.0
pyslim                  0.700
tskit                   0.4.1
```

Running the failing seed through valgrind produces the following report:
```
==2038232== Memcheck, a memory error detector
==2038232== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==2038232== Using Valgrind-3.18.1 and LibVEX; rerun with -h for copyright info
==2038232== Command: ./SLiM/bin/slim -s 22001244111718 minimal_error.slim
==2038232== 
// Initial random seed:
22001244111718

// RunInitializeCallbacks():
initializeSLiMModelType(modelType = 'nonWF');
initializeTreeSeq();
initializeMutationType(1, 0.5, "n", 0, 0.1);
initializeGenomicElementType(1, m1, 1);
initializeGenomicElement(g1, 0, 99999999);
initializeRecombinationRate(1e-08);
initializeMutationRate(1e-09);

// Starting run at generation <start>:
1 

Saving generation: 2001
Saving generation: 2101
Saving generation: 2201
Saving generation: 2301
Saving generation: 2401
Saving generation: 2501
Extinction in generation 2510.
==2038232== Invalid write of size 8
==2038232==    at 0x2C7789: SLiMSim::AddParentsColumnForOutput(tsk_table_collection_t*, robin_hood::detail::Table<true, 80ul, long, int, robin_hood::hash<long, void>, std::equal_to<long> >*) (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x2E6A49: SLiMSim::WriteTreeSequence(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, bool, bool, bool, EidosDictionaryUnretained*) (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x2FD4DA: SLiMSim::ExecuteMethod_treeSeqOutput(unsigned int, std::vector<Eidos_intrusive_ptr<EidosValue>, std::allocator<Eidos_intrusive_ptr<EidosValue> > > const&, EidosInterpreter&) (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x3109B7: SLiMSim::ExecuteInstanceMethod(unsigned int, std::vector<Eidos_intrusive_ptr<EidosValue>, std::allocator<Eidos_intrusive_ptr<EidosValue> > > const&, EidosInterpreter&) (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x53802F: EidosValue_Object_singleton::ExecuteMethodCall(unsigned int, EidosInstanceMethodSignature const*, std::vector<Eidos_intrusive_ptr<EidosValue>, std::allocator<Eidos_intrusive_ptr<EidosValue> > > const&, EidosInterpreter&) (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x484031: EidosInterpreter::Evaluate_Call(EidosASTNode const*) (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x46780E: EidosInterpreter::Evaluate_CompoundStatement(EidosASTNode const*) (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x479BA1: EidosInterpreter::Evaluate_If(EidosASTNode const*) (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x46780E: EidosInterpreter::Evaluate_CompoundStatement(EidosASTNode const*) (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x467254: EidosInterpreter::EvaluateInternalBlock(EidosScript*) (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x2804C3: Population::ExecuteScript(SLiMEidosBlock*, int, Chromosome const&) (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x2E3922: SLiMSim::_RunOneGenerationNonWF() (in /home/clw/src/SLiM/bin/slim)
==2038232==  Address 0x62ace08 is 0 bytes after a block of size 12,328 alloc'd
==2038232==    at 0x483E899: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==2038232==    by 0x2C7554: SLiMSim::AddParentsColumnForOutput(tsk_table_collection_t*, robin_hood::detail::Table<true, 80ul, long, int, robin_hood::hash<long, void>, std::equal_to<long> >*) (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x2E6A49: SLiMSim::WriteTreeSequence(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, bool, bool, bool, EidosDictionaryUnretained*) (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x2FD4DA: SLiMSim::ExecuteMethod_treeSeqOutput(unsigned int, std::vector<Eidos_intrusive_ptr<EidosValue>, std::allocator<Eidos_intrusive_ptr<EidosValue> > > const&, EidosInterpreter&) (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x3109B7: SLiMSim::ExecuteInstanceMethod(unsigned int, std::vector<Eidos_intrusive_ptr<EidosValue>, std::allocator<Eidos_intrusive_ptr<EidosValue> > > const&, EidosInterpreter&) (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x53802F: EidosValue_Object_singleton::ExecuteMethodCall(unsigned int, EidosInstanceMethodSignature const*, std::vector<Eidos_intrusive_ptr<EidosValue>, std::allocator<Eidos_intrusive_ptr<EidosValue> > > const&, EidosInterpreter&) (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x484031: EidosInterpreter::Evaluate_Call(EidosASTNode const*) (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x46780E: EidosInterpreter::Evaluate_CompoundStatement(EidosASTNode const*) (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x479BA1: EidosInterpreter::Evaluate_If(EidosASTNode const*) (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x46780E: EidosInterpreter::Evaluate_CompoundStatement(EidosASTNode const*) (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x467254: EidosInterpreter::EvaluateInternalBlock(EidosScript*) (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x2804C3: Population::ExecuteScript(SLiMEidosBlock*, int, Chromosome const&) (in /home/clw/src/SLiM/bin/slim)
==2038232== 
==2038232== Invalid read of size 8
==2038232==    at 0x5A3F48: write_table_ragged_cols (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x5BECF8: tsk_table_collection_dumpf (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x5BF12C: tsk_table_collection_dump (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x2E6C18: SLiMSim::WriteTreeSequence(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, bool, bool, bool, EidosDictionaryUnretained*) (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x2FD4DA: SLiMSim::ExecuteMethod_treeSeqOutput(unsigned int, std::vector<Eidos_intrusive_ptr<EidosValue>, std::allocator<Eidos_intrusive_ptr<EidosValue> > > const&, EidosInterpreter&) (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x3109B7: SLiMSim::ExecuteInstanceMethod(unsigned int, std::vector<Eidos_intrusive_ptr<EidosValue>, std::allocator<Eidos_intrusive_ptr<EidosValue> > > const&, EidosInterpreter&) (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x53802F: EidosValue_Object_singleton::ExecuteMethodCall(unsigned int, EidosInstanceMethodSignature const*, std::vector<Eidos_intrusive_ptr<EidosValue>, std::allocator<Eidos_intrusive_ptr<EidosValue> > > const&, EidosInterpreter&) (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x484031: EidosInterpreter::Evaluate_Call(EidosASTNode const*) (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x46780E: EidosInterpreter::Evaluate_CompoundStatement(EidosASTNode const*) (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x479BA1: EidosInterpreter::Evaluate_If(EidosASTNode const*) (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x46780E: EidosInterpreter::Evaluate_CompoundStatement(EidosASTNode const*) (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x467254: EidosInterpreter::EvaluateInternalBlock(EidosScript*) (in /home/clw/src/SLiM/bin/slim)
==2038232==  Address 0x62ace08 is 0 bytes after a block of size 12,328 alloc'd
==2038232==    at 0x483E899: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==2038232==    by 0x2C7554: SLiMSim::AddParentsColumnForOutput(tsk_table_collection_t*, robin_hood::detail::Table<true, 80ul, long, int, robin_hood::hash<long, void>, std::equal_to<long> >*) (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x2E6A49: SLiMSim::WriteTreeSequence(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, bool, bool, bool, EidosDictionaryUnretained*) (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x2FD4DA: SLiMSim::ExecuteMethod_treeSeqOutput(unsigned int, std::vector<Eidos_intrusive_ptr<EidosValue>, std::allocator<Eidos_intrusive_ptr<EidosValue> > > const&, EidosInterpreter&) (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x3109B7: SLiMSim::ExecuteInstanceMethod(unsigned int, std::vector<Eidos_intrusive_ptr<EidosValue>, std::allocator<Eidos_intrusive_ptr<EidosValue> > > const&, EidosInterpreter&) (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x53802F: EidosValue_Object_singleton::ExecuteMethodCall(unsigned int, EidosInstanceMethodSignature const*, std::vector<Eidos_intrusive_ptr<EidosValue>, std::allocator<Eidos_intrusive_ptr<EidosValue> > > const&, EidosInterpreter&) (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x484031: EidosInterpreter::Evaluate_Call(EidosASTNode const*) (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x46780E: EidosInterpreter::Evaluate_CompoundStatement(EidosASTNode const*) (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x479BA1: EidosInterpreter::Evaluate_If(EidosASTNode const*) (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x46780E: EidosInterpreter::Evaluate_CompoundStatement(EidosASTNode const*) (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x467254: EidosInterpreter::EvaluateInternalBlock(EidosScript*) (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x2804C3: Population::ExecuteScript(SLiMEidosBlock*, int, Chromosome const&) (in /home/clw/src/SLiM/bin/slim)
==2038232== 
==2038232== Invalid read of size 8
==2038232==    at 0x5A401B: write_table_ragged_cols (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x5BECF8: tsk_table_collection_dumpf (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x5BF12C: tsk_table_collection_dump (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x2E6C18: SLiMSim::WriteTreeSequence(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, bool, bool, bool, EidosDictionaryUnretained*) (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x2FD4DA: SLiMSim::ExecuteMethod_treeSeqOutput(unsigned int, std::vector<Eidos_intrusive_ptr<EidosValue>, std::allocator<Eidos_intrusive_ptr<EidosValue> > > const&, EidosInterpreter&) (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x3109B7: SLiMSim::ExecuteInstanceMethod(unsigned int, std::vector<Eidos_intrusive_ptr<EidosValue>, std::allocator<Eidos_intrusive_ptr<EidosValue> > > const&, EidosInterpreter&) (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x53802F: EidosValue_Object_singleton::ExecuteMethodCall(unsigned int, EidosInstanceMethodSignature const*, std::vector<Eidos_intrusive_ptr<EidosValue>, std::allocator<Eidos_intrusive_ptr<EidosValue> > > const&, EidosInterpreter&) (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x484031: EidosInterpreter::Evaluate_Call(EidosASTNode const*) (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x46780E: EidosInterpreter::Evaluate_CompoundStatement(EidosASTNode const*) (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x479BA1: EidosInterpreter::Evaluate_If(EidosASTNode const*) (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x46780E: EidosInterpreter::Evaluate_CompoundStatement(EidosASTNode const*) (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x467254: EidosInterpreter::EvaluateInternalBlock(EidosScript*) (in /home/clw/src/SLiM/bin/slim)
==2038232==  Address 0x62ace08 is 0 bytes after a block of size 12,328 alloc'd
==2038232==    at 0x483E899: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==2038232==    by 0x2C7554: SLiMSim::AddParentsColumnForOutput(tsk_table_collection_t*, robin_hood::detail::Table<true, 80ul, long, int, robin_hood::hash<long, void>, std::equal_to<long> >*) (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x2E6A49: SLiMSim::WriteTreeSequence(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, bool, bool, bool, EidosDictionaryUnretained*) (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x2FD4DA: SLiMSim::ExecuteMethod_treeSeqOutput(unsigned int, std::vector<Eidos_intrusive_ptr<EidosValue>, std::allocator<Eidos_intrusive_ptr<EidosValue> > > const&, EidosInterpreter&) (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x3109B7: SLiMSim::ExecuteInstanceMethod(unsigned int, std::vector<Eidos_intrusive_ptr<EidosValue>, std::allocator<Eidos_intrusive_ptr<EidosValue> > > const&, EidosInterpreter&) (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x53802F: EidosValue_Object_singleton::ExecuteMethodCall(unsigned int, EidosInstanceMethodSignature const*, std::vector<Eidos_intrusive_ptr<EidosValue>, std::allocator<Eidos_intrusive_ptr<EidosValue> > > const&, EidosInterpreter&) (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x484031: EidosInterpreter::Evaluate_Call(EidosASTNode const*) (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x46780E: EidosInterpreter::Evaluate_CompoundStatement(EidosASTNode const*) (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x479BA1: EidosInterpreter::Evaluate_If(EidosASTNode const*) (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x46780E: EidosInterpreter::Evaluate_CompoundStatement(EidosASTNode const*) (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x467254: EidosInterpreter::EvaluateInternalBlock(EidosScript*) (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x2804C3: Population::ExecuteScript(SLiMEidosBlock*, int, Chromosome const&) (in /home/clw/src/SLiM/bin/slim)
==2038232== 
==2038232== Syscall param write(buf) points to uninitialised byte(s)
==2038232==    at 0x4CDF907: write (in /usr/lib/libc-2.33.so)
==2038232==    by 0x4C709CC: _IO_file_write@@GLIBC_2.2.5 (in /usr/lib/libc-2.33.so)
==2038232==    by 0x4C6FD45: new_do_write (in /usr/lib/libc-2.33.so)
==2038232==    by 0x4C71A68: _IO_do_write@@GLIBC_2.2.5 (in /usr/lib/libc-2.33.so)
==2038232==    by 0x4C71015: _IO_file_xsputn@@GLIBC_2.2.5 (in /usr/lib/libc-2.33.so)
==2038232==    by 0x4C65A80: fwrite (in /usr/lib/libc-2.33.so)
==2038232==    by 0x5A1794: kastore_close (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x5BF04D: tsk_table_collection_dumpf (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x5BF12C: tsk_table_collection_dump (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x2E6C18: SLiMSim::WriteTreeSequence(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, bool, bool, bool, EidosDictionaryUnretained*) (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x2FD4DA: SLiMSim::ExecuteMethod_treeSeqOutput(unsigned int, std::vector<Eidos_intrusive_ptr<EidosValue>, std::allocator<Eidos_intrusive_ptr<EidosValue> > > const&, EidosInterpreter&) (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x3109B7: SLiMSim::ExecuteInstanceMethod(unsigned int, std::vector<Eidos_intrusive_ptr<EidosValue>, std::allocator<Eidos_intrusive_ptr<EidosValue> > > const&, EidosInterpreter&) (in /home/clw/src/SLiM/bin/slim)
==2038232==  Address 0x69cd320 is 3,072 bytes inside a block of size 4,096 alloc'd
==2038232==    at 0x483E899: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==2038232==    by 0x4C64563: _IO_file_doallocate (in /usr/lib/libc-2.33.so)
==2038232==    by 0x4C72DAF: _IO_doallocbuf (in /usr/lib/libc-2.33.so)
==2038232==    by 0x4C71F67: _IO_file_overflow@@GLIBC_2.2.5 (in /usr/lib/libc-2.33.so)
==2038232==    by 0x4C71015: _IO_file_xsputn@@GLIBC_2.2.5 (in /usr/lib/libc-2.33.so)
==2038232==    by 0x4C65A80: fwrite (in /usr/lib/libc-2.33.so)
==2038232==    by 0x5A14E0: kastore_close (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x5BF04D: tsk_table_collection_dumpf (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x5BF12C: tsk_table_collection_dump (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x2E6C18: SLiMSim::WriteTreeSequence(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, bool, bool, bool, EidosDictionaryUnretained*) (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x2FD4DA: SLiMSim::ExecuteMethod_treeSeqOutput(unsigned int, std::vector<Eidos_intrusive_ptr<EidosValue>, std::allocator<Eidos_intrusive_ptr<EidosValue> > > const&, EidosInterpreter&) (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x3109B7: SLiMSim::ExecuteInstanceMethod(unsigned int, std::vector<Eidos_intrusive_ptr<EidosValue>, std::allocator<Eidos_intrusive_ptr<EidosValue> > > const&, EidosInterpreter&) (in /home/clw/src/SLiM/bin/slim)
==2038232== 
==2038232== Syscall param write(buf) points to uninitialised byte(s)
==2038232==    at 0x4CDF907: write (in /usr/lib/libc-2.33.so)
==2038232==    by 0x4C709CC: _IO_file_write@@GLIBC_2.2.5 (in /usr/lib/libc-2.33.so)
==2038232==    by 0x4C6FD45: new_do_write (in /usr/lib/libc-2.33.so)
==2038232==    by 0x4C7109D: _IO_file_xsputn@@GLIBC_2.2.5 (in /usr/lib/libc-2.33.so)
==2038232==    by 0x4C65A80: fwrite (in /usr/lib/libc-2.33.so)
==2038232==    by 0x5A1794: kastore_close (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x5BF04D: tsk_table_collection_dumpf (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x5BF12C: tsk_table_collection_dump (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x2E6C18: SLiMSim::WriteTreeSequence(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, bool, bool, bool, EidosDictionaryUnretained*) (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x2FD4DA: SLiMSim::ExecuteMethod_treeSeqOutput(unsigned int, std::vector<Eidos_intrusive_ptr<EidosValue>, std::allocator<Eidos_intrusive_ptr<EidosValue> > > const&, EidosInterpreter&) (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x3109B7: SLiMSim::ExecuteInstanceMethod(unsigned int, std::vector<Eidos_intrusive_ptr<EidosValue>, std::allocator<Eidos_intrusive_ptr<EidosValue> > > const&, EidosInterpreter&) (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x53802F: EidosValue_Object_singleton::ExecuteMethodCall(unsigned int, EidosInstanceMethodSignature const*, std::vector<Eidos_intrusive_ptr<EidosValue>, std::allocator<Eidos_intrusive_ptr<EidosValue> > > const&, EidosInterpreter&) (in /home/clw/src/SLiM/bin/slim)
==2038232==  Address 0x55647d8 is 1,128 bytes inside a block of size 36,984 alloc'd
==2038232==    at 0x483E899: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==2038232==    by 0x5A271D: kastore_puts (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x5A3F23: write_table_ragged_cols (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x5BECF8: tsk_table_collection_dumpf (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x5BF12C: tsk_table_collection_dump (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x2E6C18: SLiMSim::WriteTreeSequence(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, bool, bool, bool, EidosDictionaryUnretained*) (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x2FD4DA: SLiMSim::ExecuteMethod_treeSeqOutput(unsigned int, std::vector<Eidos_intrusive_ptr<EidosValue>, std::allocator<Eidos_intrusive_ptr<EidosValue> > > const&, EidosInterpreter&) (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x3109B7: SLiMSim::ExecuteInstanceMethod(unsigned int, std::vector<Eidos_intrusive_ptr<EidosValue>, std::allocator<Eidos_intrusive_ptr<EidosValue> > > const&, EidosInterpreter&) (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x53802F: EidosValue_Object_singleton::ExecuteMethodCall(unsigned int, EidosInstanceMethodSignature const*, std::vector<Eidos_intrusive_ptr<EidosValue>, std::allocator<Eidos_intrusive_ptr<EidosValue> > > const&, EidosInterpreter&) (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x484031: EidosInterpreter::Evaluate_Call(EidosASTNode const*) (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x46780E: EidosInterpreter::Evaluate_CompoundStatement(EidosASTNode const*) (in /home/clw/src/SLiM/bin/slim)
==2038232==    by 0x479BA1: EidosInterpreter::Evaluate_If(EidosASTNode const*) (in /home/clw/src/SLiM/bin/slim)
==2038232== 
==2038232== 
==2038232== HEAP SUMMARY:
==2038232==     in use at exit: 20,342,817 bytes in 13,247 blocks
==2038232==   total heap usage: 132,089 allocs, 118,842 frees, 292,232,709 bytes allocated
==2038232== 
==2038232== LEAK SUMMARY:
==2038232==    definitely lost: 284,512 bytes in 3,940 blocks
==2038232==    indirectly lost: 1,136,155 bytes in 3,574 blocks
==2038232==      possibly lost: 18,320 bytes in 8 blocks
==2038232==    still reachable: 18,903,830 bytes in 5,725 blocks
==2038232==                       of which reachable via heuristic:
==2038232==                         newarray           : 53,184 bytes in 740 blocks
==2038232==         suppressed: 0 bytes in 0 blocks
==2038232== Rerun with --leak-check=full to see details of leaked memory
==2038232== 
==2038232== Use --track-origins=yes to see where uninitialised values come from
==2038232== For lists of detected and suppressed errors, rerun with: -s
==2038232== ERROR SUMMARY: 6 errors from 5 contexts (suppressed: 0 from 0)
```

**Comment by bhaller** [comment]
Thanks for the awesome bug report!  I'll look into it when I'm back from vacation in early February, unless @petrelharp gets there first.  :->

**Comment by petrelharp** [comment]
Yes, wonderful report!  I won't get to it *today*...

How often is "occasional", btw, @clwgg?

**Comment by clwgg** [comment]
I haven't yet quantified it really, but while cutting down the script and testing if the failure still occurred after removing certain features, I could usually get it to fail within 3-5 runs on random seeds.

**Comment by bhaller** [comment]
Sorry for the delay in fixing this, things are a bit busy right now.  :->  I believe that I understand the problem, and the fix ought to be good.  Let me know if you see any further problems.  Thanks for the excellent bug report, which made finding the bug very easy!

**Comment by clwgg** [comment]
Wonderful news! I can confirm that I don't seem to encounter the failure on my end anymore. Thank you!

**Comment by petrelharp** [comment]
... and, this bug just hit us - I assume, since we had replicable core dumps on tree seq output that went away when we switched to development SLiM. Thanks, all!

**Comment by bhaller** [comment]
So, this fix is impetus to roll a 3.7.1 release, I suppose; this crash is pretty bad.  @petrelharp there was a stdpopsim issue that you guys wanted a roll for, right?  I forget the details of that, though.  It got fixed already, didn't it?  Sorry, a bit disconnected over here, juggling some other stuff; but maybe I should roll out 3.7.1?

**Comment by petrelharp** [comment]
Yes - it was #274, closed already. And yes, that'd be awesome - these two are good reasons to roll out a release.
